lib/wda/xcodebuild.js

```diff
@@ -56,6 +58,14 @@ class XcodeBuild {
   async init (noSessionProxy) {
     this.noSessionProxy = noSessionProxy;
 
+    if (this.useXCTESTRUNFile) {
+      if (this.xcodeVersion.major > 7) {
```

Comment:
I'd prefer this condition switched, so error handling is taken care of before processing:
```js
if (this.xcodeVersion.major <= 7) {
  log.errorAndThrow('useXCTESTRUNFile can only be used with xcode version 8 onwards');
}
this.xctestrunFilePath = await setXCTESTRUNFile(this.realDevice, this.device.udid, this.platformVersion, this.bootstrapPath, this.wdaRemotePort);
return;
```